:param v2

huboneo opened the pull request:
This PR adds support for the `:param` v2 syntax (explicit returns). In addition, it also cleans up and standardizes the error frame appearance regardless of error origin. Finally it standardizes the appearance of `:help` buttons.

### Todo
- [x] Fix tests once `@neo4j/browser-lambda-parser` is published

### STR
- run `:help param` and follow the instructions for explicit returns

### Screenshots
Setting parameters
<img width="1492" alt="Screenshot 2019-08-27 at 15 45 57" src="https://user-images.githubusercontent.com/52443771/63776486-c8964980-c8e1-11e9-956b-696f3c2e1f8e.png">

Error
<img width="1496" alt="Screenshot 2019-08-27 at 14 31 15" src="https://user-images.githubusercontent.com/52443771/63776586-f8455180-c8e1-11e9-9991-795628f4425f.png">

`:help` commands
<img width="1496" alt="Screenshot 2019-08-27 at 15 43 16" src="https://user-images.githubusercontent.com/52443771/63776607-fe3b3280-c8e1-11e9-81f2-9e0881a88642.png">

huboneo commented on src/shared/modules/commands/helpers/lambdas.js:
```diff
@@ -0,0 +1,152 @@
+      }
+
+      // future proofing
+      if (parameters.type !== ARRAY) return null
```
maybe return `{}`

oskarhane commented on package.json:
```diff
@@ -123,6 +123,7 @@
     "xml2js": "^0.4.19"
   },
   "dependencies": {
+    "@neo4j/browser-lambda-parser": "^1.0.2",
```
🎉

oskarhane commented on src/browser/modules/Help/html/param.html:
```diff
@@ -18,6 +18,12 @@
+        <br/> Explicit returns yield a list of records, matching that of your Cypher query: <code>x => { RETURN 1 as foo }</code> yields <code>$x = [{foo: 1}]</code>
+        <br/> You can pick out individual values from your result using destructuring: <code> [{foo}] => { RETURN 1 as foo }</code> yields <code>$foo = 1</code>
+        <br/> You can also rename destructured params: <code> [{foo: bar}] => { RETURN 1 as foo }</code> yields <code>$bar = 1</code>
+      </p>
```
👍

oskarhane commented on src/browser/modules/Stream/auto-exec-button.jsx:
```diff
@@ -0,0 +1,65 @@
+  outline: transparent;
+`
+
+function AutoExecButtonComponent ({ bus, cmd, cmdChar }) {
```
Could you add tests for this component, to make sure what we expect is send over the bus, and that the rendered output is what's expected?
We probably have it somewhat covered in the e2e tests, but it's faster feedback if it breaks already here if we introduce a regression in the future.

huboneo commented:
@oskarhane thank you for your feedback! I have addressed your comments. For the AutoExecButton tests I had to export a non-bus version as I could not figure out how to properly mock `react-suber` Please let me know if you have a better way
